Sources/StreamChatUI/ChatChannelList/ChatChannelAvatarView_Tests.swift

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import StreamChatTestTools
+@testable import StreamChatUI
+import XCTest
+
+class ChatChannelAvatarView_Tests: XCTestCase {
+    var currentUserId: UserId!
+    var channel: ChatChannel!
+    
+    override func setUp() {
+        super.setUp()
+        currentUserId = .unique
+
+        channel = ChatChannel.mockDMChannel(members: [
+            .mock(id: currentUserId, imageURL: TestImages.vader.url),
+            .mock(id: .unique, imageURL: TestImages.yoda.url, isOnline: true)
+        ])
+    }
+    
+    func test_emptyAppearance() {
+        let view = ChatChannelAvatarView().withoutAutoresizingMaskConstraints
+        view.addSizeConstraints()
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_defaultAppearance_withDirectMessageChannel() {
+        let view = ChatChannelAvatarView().withoutAutoresizingMaskConstraints
+        view.addSizeConstraints()
+        view.content = (channel, currentUserId)
+        AssertSnapshot(view, variants: [.small], suffix: "with online indicator")
+
+        // Reset the channel such that both members are offline
+        channel = ChatChannel.mockDMChannel(members: [
+            .mock(id: currentUserId, imageURL: TestImages.vader.url),
+            .mock(id: .unique, imageURL: TestImages.yoda.url)
+        ])
+
+        view.content = (channel, currentUserId)
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_defaultAppearance_withNonDMChannel() {
+        // TODO: https://stream-io.atlassian.net/browse/CIS-652
+    }
+
+    func test_appearanceCustomization_usingUIConfig() {
+        class RectIndicator: UIView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .systemPink
+                // swiftlint:disable:next constraints_forbidden
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        var config = UIConfig()
+        config.channelList.channelListItemSubviews.onlineIndicator = RectIndicator.self
+        config.colorPalette.alternativeActiveTint = .brown
+        config.colorPalette.lightBorder = .cyan
+
+        let view = ChatChannelAvatarView().withoutAutoresizingMaskConstraints
+        view.addSizeConstraints()
+        view.uiConfig = config
+        view.content = (channel, currentUserId)
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_appearanceCustomization_usingAppearanceHook() {
+        class TestView: ChatChannelAvatarView {}
+        TestView.defaultAppearance {
+            // Modify appearance
+            $0.onlineIndicatorView.backgroundColor = .red
+            $0.backgroundColor = .yellow
+
+            // Modify layout
+            NSLayoutConstraint.activate([
+                // swiftlint:disable constraints_forbidden
```

Comment:
Maybe better to disable this rule for UI_Tests ?

Reply:
Yeah, I was thinking about it but wasn't sure. It probably makes sense. Implemented in 26f2db2